bundles/org.openhab.core.audio/src/main/java/org/openhab/core/audio/AudioSink.java

```diff
@@ -58,13 +58,47 @@ public interface AudioSink {
      *
      * In case the audioStream is null, this should be interpreted as a request to end any currently playing stream.
      *
+     * When the process method ends, if the stream implements the {@link org.openhab.core.common.Disposable} interface,
+     * the sink should hereafter get rid of it by calling the dispose method.
+     *
      * @param audioStream the audio stream to play or null to keep quiet
      * @throws UnsupportedAudioFormatException If audioStream format is not supported
      * @throws UnsupportedAudioStreamException If audioStream is not supported
      */
     void process(@Nullable AudioStream audioStream)
             throws UnsupportedAudioFormatException, UnsupportedAudioStreamException;
 
+    /**
+     * Processes the passed {@link AudioStream}, and executes the whenFinished Runnable (or stores for later execution
+     * if the sink is asynchronous).
+     *
+     * If the passed {@link AudioStream} is not supported by this instance, an {@link UnsupportedAudioStreamException}
+     * is thrown.
+     *
+     * If the passed {@link AudioStream} has an {@link AudioFormat} not supported by this instance,
+     * an {@link UnsupportedAudioFormatException} is thrown.
+     *
+     * In case the audioStream is null, this should be interpreted as a request to end any currently playing stream.
+     *
+     * When the process method ends, if the stream implements the {@link org.openhab.core.common.Disposable} interface,
+     * the sink should hereafter get rid of it by calling the dispose method.
+     *
+     * @param audioStream the audio stream to play or null to keep quiet
+     * @param whenFinished A Runnable to run when the sound is finished playing.
+     * @throws UnsupportedAudioFormatException If audioStream format is not supported
+     * @throws UnsupportedAudioStreamException If audioStream is not supported
+     */
+    default void process(@Nullable AudioStream audioStream, @Nullable Runnable whenFinished)
```

Comment:
OK, I rewrote it using CompletableFuture.
Summary of changes around this :
- In one place I had to return a record (ServedStream) instead of just a String in order to return two informations : the CompletableFuture and the stream url, instead of only the stream URL. ServedStream was previously private and is now public.
- Returning a CompletableFuture instead of passing a Runnable forced me to change the method name to avoid collision.
And changing the method signature/name made me think that the old process method should not be used and returning a CompletableFuture should be encouraged by deprecating the old one. Is it OK ?